### DISCUSSION TITLE
Global objects mentioned in lang.java configuration instructions are not recognized.

### DISCUSSION THREAD

**Discussion by rasivasu** [LazyVim/LazyVim#1618] [Q&A]
I am using https://www.lazyvim.org/extras/lang/java for configuration Java Language using Lazy Vim.  

When I copy nvim-jdtls configuration into the newly created file under lua/plugins, the following global objects are not recognized.

- java_filetypes
- Utils
- extend_or_override

This also results in run time error when working with Java files.

**Comment by folke** [comment]
You should **NOT** copy that code!

Read the first paragraph of the page you linked to.

The options below that are just a reference of the defaults.

**Reply by rasivasu** [reply]
@folke Thanks.

I found that adding the code at the bottom of this message resolves all the issues.

```
local Util = require("lazyvim.util")

-- This is the same as in lspconfig.server_configurations.jdtls, but avoids
-- needing to require that when this module loads.
local java_filetypes = { "java" }

-- Utility function to extend or override a config table, similar to the way
-- that Plugin.opts works.
---@param config table
---@param custom function | table | nil
local function extend_or_override(config, custom, ...)
  if type(custom) == "function" then
    config = custom(config, ...) or config
  elseif custom then
    config = vim.tbl_deep_extend("force", config, custom) --[[@as table]]
  end
  return config
end

```
